app/services/clients/adminusers_client.js

```diff
@@ -4,53 +4,62 @@
 const baseClient = require('./base_client/base_client')
 const requestLogger = require('../../utils/request_logger')
 const Service = require('../../models/Service.class')
-const { createCallbackToPromiseConverter } = require('../../utils/response_converter')
+const { getCounter } = require('../../metrics/graphite_reporter')
 
 // Constants
 const SERVICE_NAME = 'adminusers'
-
-const responseBodyToServiceTransformer = function responseBodyToServiceTransformer (body) {
-  try {
-    const service = new Service(body)
-    return Promise.resolve(service)
-  } catch (error) {
-    return Promise.reject(error)
-  }
-}
+const METRICS_PREFIX = 'internal-rest-call.adminusers'
+const SUCCESS_CODES = [200, 201, 202, 204, 206]
 
 let baseUrl
 let correlationId
 
-const findServiceBy = function findServiceBy (findOptions, loggingFields = {}) {
+/** @private */
+const _getAdminUsers = (url, description, findOptions, loggingFields = {}, callingFunction) => {
   return new Promise(function (resolve, reject) {
-    const servicesResource = `${baseUrl}/v1/api/services`
-    const params = {
-      correlationId: correlationId,
-      qs: {
-        gatewayAccountId: findOptions.gatewayAccountId
-      }
-    }
     const startTime = new Date()
-
-    // @FIXME(sfount) we should NOT be passing the resolve/ reject of a promise
-    //                down through 'cotext' - this tightly couples the code and
-    //                gives us no control of when our code handles errors
     const context = {
-      url: servicesResource,
-      defer: { resolve, reject },
+      url: url,
       startTime: startTime,
-      correlationId: correlationId,
       method: 'GET',
-      description: 'find a service',
+      description: description,
       service: SERVICE_NAME
     }
-    const callbackToPromiseConverter = createCallbackToPromiseConverter(context, responseBodyToServiceTransformer, loggingFields)
     requestLogger.logRequestStart(context, loggingFields)
-    baseClient.get(servicesResource, params, callbackToPromiseConverter)
-      .on('error', callbackToPromiseConverter)
+    const params = {
+      correlationId: correlationId,
+      qs: {
+        gatewayAccountId: findOptions.gatewayAccountId
+      }
+    }
+    baseClient.get(url, params, null)
+    .then(response => {
```

Comment:
Now that this method is promise based there's no need for `_getAdminUsers` to return a new Promise, we found doing this can make the code harder to read and is more error prone. 

Returning this promise will behave in the same was as constructing a new promise here. 

Something like:
 
```javascript
function _getAdminUsers() {
  return baseClient.get()
    .then((respone) => {
      return response.body
    })
    .catch((error) => {
      
      // ...
      throw error
    })
} 
```